modules/ingestor/src/service/format.rs

```diff
@@ -92,19 +97,33 @@ impl<'g> Format {
                 let curation: Curation = serde_yml::from_slice(&buffer)?;
                 loader.load(labels, curation, digests).await
             }
+            Format::CweCatalog => {
+                let loader = CweCatalogLoader::new(graph);
+                loader.load_bytes(labels, &buffer, digests).await
+            }
+        }
+    }
+
+    pub fn is_format(&self, bytes: &[u8]) -> Result<bool, Error> {
+        match self {
+            Format::OSV => Self::is_osv(bytes),
+            Format::CSAF => Self::is_csaf(bytes),
+            Format::CVE => Self::is_cve(bytes),
+            Format::SPDX => Self::is_spdx(bytes),
+            Format::CycloneDX => Self::is_cyclonedx(bytes),
+            Format::ClearlyDefined => Self::is_clearly_defined(bytes),
+            Format::CweCatalog => Self::is_cwe_catalog(bytes),
         }
     }
 
     pub fn from_bytes(bytes: &[u8]) -> Result<Self, Error> {
-        match Self::advisory_from_bytes(bytes) {
-            Err(Error::UnsupportedFormat(ea)) => match Self::sbom_from_bytes(bytes) {
-                Err(Error::UnsupportedFormat(es)) => {
-                    Err(Error::UnsupportedFormat(format!("{ea}\n{es}")))
-                }
-                x => x,
-            },
-            x => x,
+        for format in Self::iter() {
+            if format.is_format(bytes)? {
+                return Ok(format);
+            }
```

Comment:
I don't love this. I don't trust the iteration order, and I think (in the spirit of @ctron 's concern above) that we should be intentional about the order of our format checks. The checks are fast (and when metrics reveal that to be false, easily parallelizable) but we should check for the ones we expect more frequently first. Hence, my divvying the logic up between advisories and SBOM's, as I expect to see far more of the former than the latter. Similarly we should check for the more common formats first.

Reply:
I believe `strum` does assert that the `EnumIter` is in definition order, so we can put "expensive" ones last.

Talking to @ctron, I also agree, iff we know (e.g., from an importer) then pass directly, and reserve the "guess" case for "user is dragging/dropping a file into the browser".

Reply:
I disagree that rearranging the order of the variants and assuming that strum honors that order is more obvious than [the implementations of `advisory_from_bytes` and `sbom_from_bytes`](https://github.com/trustification/trustify/blob/6f0a558be9d4fe78c51eba550b6218b83e78d878/modules/ingestor/src/service/format.rs#L110-L136).

Reply:
> Talking to @ctron, I also agree, iff we know (e.g., from an importer) then pass directly, and reserve the "guess" case for "user is dragging/dropping a file into the browser".

Sure, adding an `Option<Format>` ingest argument makes sense.

Reply:
Either way, can we move that problem to a different subsequent PR?

Reply:
> Either way, can we move that problem to a different subsequent PR?

Yes

Reply:
Yah, I think keep the guessing for the human-centric usecase, but the CSAF Importer is pretty sure it's CSAF.  The OSV importer knows it's OSV, etc.